### PR TITLE
[heli][flight_plans] Heli Spinup routine and flight plan mode manual

### PR DESCRIPTION
**Pull request by fvantienen**
- Functionality for mode "manual" in flight plan without stabilization
- Navigation routines for spinning up rotor for helicopters in flight plan
- Added example flightplan

### CODE REVIEW

**Comment by flixr** [comment]
Personally I'm not in favour of adding ever more functions that always return false only to be able to `call` them from the flight plan without `loop="false"`.
See also old discussion about this: https://github.com/paparazzi/paparazzi/pull/830

**Comment by fvantienen** [comment]
Aha yes I forgot that one already.. will change that small thing ;)

File: sw/airborne/firmwares/rotorcraft/navigation.c [hunk 1]
```diff
@@ -480,6 +481,15 @@ void nav_home(void)
   nav_run();
 }
 
+/** Set manual roll, pitch and yaw without stabilization */
```
Comment (flixr) [inline sw/airborne/firmwares/rotorcraft/navigation.c]: please add what unit the arguments are supposed to be in

File: sw/airborne/modules/nav/nav_heli_spinup.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "modules/nav/nav_heli_spinup.h"
+#include "navigation.h"
+#include "paparazzi.h"
+
+static struct nav_heli_spinup_t nav_heli_spinup;
+
+bool nav_heli_spinup_setup(uint16_t duration, float throttle)
+{
+  nav_heli_spinup.duration = duration;
```
Comment (flixr) [inline sw/airborne/modules/nav/nav_heli_spinup.c]: check that duration is not zero!

**Comment by flixr** [comment]
Please try to not put different features/additions in the same commit...

**Comment by flixr** [comment]
In flight plan: why allow different vmodes in manual instead of requiring throttle?

**Comment by flixr** [comment]
why add a FMODE command to `stabilization_cmd`?

**Comment by fvantienen** [comment]
- I didn't find the time to split up the commits, feel free to split it up into 2 different PR's
- The `vmode` is like this, because this was the easiest way to implement it and it works exactly the same as `attitude`.
- I added the `FMODE` command because eventually either the stabilization or the outer loop should set the correct throttle curve.

File: sw/airborne/firmwares/rotorcraft/navigation.h [hunk 3]
```diff
@@ -48,10 +48,11 @@ extern uint8_t horizontal_mode;
 extern struct EnuCoor_i nav_segment_start, nav_segment_end;
 extern struct EnuCoor_i nav_circle_center;
 extern int32_t nav_circle_radius, nav_circle_qdr, nav_circle_radians;
-#define HORIZONTAL_MODE_WAYPOINT  0
```
Comment (flixr) [inline sw/airborne/firmwares/rotorcraft/navigation.h]: this means changing the meaning of the already existing modes, probably better to append the manual mode at the end.
Also needs to be updated in `server_global.ml` accordingly

File: sw/airborne/firmwares/rotorcraft/navigation.c [hunk 4]
```diff
@@ -479,6 +479,20 @@ void nav_home(void)
   nav_run();
 }
 
+/** Set manual roll, pitch and yaw without stabilization
+ *
+ * @param[in] roll The amount of roll between -1.0 and 1.0
```
Comment (flixr) [inline sw/airborne/firmwares/rotorcraft/navigation.c]: really? I don't think so...
the nav_roll/pitch/heading should be in radians in fixed point

**Comment by fvantienen** [comment]
For now I changes it to `commands` instead of `stabilized_cmd`, and if I find a nicer solution later we will make a separate PR.

File: conf/modules/nav_heli_spinup.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="nav_heli_spinup" dir="nav">
+  <doc>
+    <description>This module controls the spinup of the main rotor from a Helicopter</description>
```
Comment (flixr) [inline conf/modules/nav_heli_spinup.xml]: Please try to remember that the first line of the description is taken as name (very brief descrption) in the generated http://docs.paparazziuav.org/latest/onboard_modules.html

Maybe we should really introduce a `<brief>Foo module</brief>` or so?

**Comment by flixr** [comment]
Looks good to me... @gautierhattenberger if the manual mode is fine with you as well, feel free to merge.

**Comment by fvantienen** [comment]
@gautierhattenberger any feedback?

**Comment by gautierhattenberger** [comment]
It seems that if someone tries to use this with a fixed-wing it will fail without proper error message.
I suggest to at least hide the function `nav_set_manual` in a macro like `NavSetManual` that triggers a `#error` message at compilation for FW firmware.

**Comment by fvantienen** [comment]
Is this ok @gautierhattenberger ?

**Pull request merged**